notary-commons/src/main/kotlin/sidechain/iroha/consumer/IrohaConsumerImpl.kt

```diff
@@ -42,9 +63,16 @@ class IrohaConsumerImpl(
      */
     override fun send(tx: TransactionOuterClass.Transaction): Result<String, Exception> {
         return Result.of {
-            irohaAPI.transactionSync(tx)
-        }.flatMap {
-            checkTransactionStatus(Utils.hash(tx))
+            val txStatus = TxStatus.createEmpty()
+            irohaAPI.transaction(tx, waitForTerminalStatus)
+                .blockingSubscribe(createTxStatusObserver(txStatus))
+            if (txStatus.txHash != null) {
+                txStatus.txHash!!
```

Comment:
Isn't kotlin smart enough to smart cast it without `!!`?

Reply:
Unfortunately, no. `txHash` is `var`, meaning that it may be changed to `null` any time